### ISSUE TITLE
ConvNet FP8 support

### ISSUE THREAD

**Issue by AnnaTrainingG** [NVIDIA/TensorRT#4461]
import torch
import torchvision
import modelopt.torch.quantization as mtq
_default_disabled_quantizer_cfg = {
    "nn.BatchNorm1d": {"*": {"enable": False}},
    "nn.BatchNorm2d": {"*": {"enable": False}},
    "nn.BatchNorm3d": {"*": {"enable": False}},
    "nn.LeakyReLU": {"*": {"enable": False}},
    "*lm_head*": {"enable": False},
    "*proj_out.*": {"enable": False},  # In Whisper model, lm_head has key name proj_out
    "*block_sparse_moe.gate*": {"enable": False},  # Skip the MOE router
    "*router*": {"enable": False},  # Skip the MOE router
    "*mlp.gate.*": {"enable": False},  # Skip the MOE router
    "*mlp.shared_expert_gate.*": {"enable": False},  # Skip the MOE router
    "*output_layer*": {"enable": False},
    "output.*": {"enable": False},
    "default": {"enable": False},
}
FP8_DEFAULT_CFG = {
    "quant_cfg": {
        "*weight_quantizer": {"num_bits": (4, 3), "axis": None},
        "*input_quantizer": {"num_bits": (4, 3), "axis": None},
        "*output_quantizer": {"enable": False},
        **_default_disabled_quantizer_cfg,
    },
    "algorithm": "max",
}

def calib_loop():
    for _ in range(10):
        model(torch.randn(16, 3, 224, 224, device='cuda'))

dynamic_axes = {'input': {0: 'batch'}, 'output': {0: 'batch'}}
model = torchvision.models.resnet18(pretrained=True).cuda()
mtq.quantize(model,  FP8_DEFAULT_CFG, forward_loop=calib_loop)
data = torch.randn(16,3,224,224,device='cuda')
model.forward(data) #torch.randn(16,3,224,224,device='cuda'))
print("hahahahahahah", model)
def generate_fp8_scales(unet):
    # temporary solution due to a known bug in torch.onnx._dynamo_export
    for _, module in unet.named_modules():
        if isinstance(module, (torch.nn.Linear, torch.nn.Conv2d)):
            module.input_quantizer._num_bits = 8
            module.weight_quantizer._num_bits = 8
            module.input_quantizer._amax = (module.input_quantizer._amax * 127) / 448.0
            module.weight_quantizer._amax = (module.weight_quantizer._amax * 127) / 448.0

generate_fp8_scales(model)
torch.onnx.export(
    model,
    data, # torch.randn(16, 3, 224, 224, device='cuda'),
    'resnet18_fp8_no_bn.onnx',
    opset_version=17,
    do_constant_folding=True
)
I find that the Conv+bn can't fused with relu and  Conv+bn Kernel's output type  always is FP32, very slow, slower than FP16 and int8

**Comment by nvyihengz** [comment]
Can you share the onnx model exported and what GPU was used?

**Comment by AnnaTrainingG** [comment]
Hi, you can run this code on your gpu, you will get the onnx,  I used H20 GPU

**Comment by lix19937** [comment]
You can ref  https://docs.nvidia.com/deeplearning/tensorrt/10.10.0/performance/best-practices.html#example-workflow-fp8-mha-fusion
